File: Luminal/Luminal/OpenGL/GLVertexArrayObject.cs
using OpenTK.Graphics.OpenGL;

namespace Luminal.OpenGL
{
    public class GLVertexArrayObject
    {
        public int GLObject;

        public GLVertexArrayObject()
        {
            GLObject = GL.GenVertexArray();
        }

        public GLVertexArrayObject(string name)
        {
            GLObject = GL.GenVertexArray();
            GLHelper.LabelObj(ObjectLabelIdentifier.VertexArray, GLObject, $"VAO: {name}");
        }

        public void Bind()
        {
            GL.BindVertexArray(GLObject);
        }
    }
}

File: LuminalCPP/GLVertexArrayObject.cpp
#include "GLVertexArrayObject.h"
#include <gl/glew.h>

namespace Luminal
{
    void GLVertexArrayObject::Create()
    {
        glCreateVertexArrays(1, &GLObject);
    }

    void GLVertexArrayObject::Bind()
    {
        glBindVertexArray(GLObject);
    }
}